### DISCUSSION TITLE
username vs email issue

### DISCUSSION THREAD

**Discussion by JamesM92** [docker-mailserver/docker-mailserver#2736] [Q&A]
Trying to troubleshoot an issue, and im not sure whats causing it.

I'm sending an email from my admin account to an alias of my admin account. I.E. sending an email to myself.

but the email is getting rejected from the server, because from what i can tell its trying to redirect it to the username "admin" while the server is expecting admin@penguinarmy.club.

I'm not sure if this is an issue with thunderbird as it is similar to the login issue I've had earlier, but I feel like this is internal name issue?

![image](https://user-images.githubusercontent.com/36548759/185795411-f1845eb5-c680-4c63-9e35-d3c7651f813e.png)

```

Aug 21 09:00:32 penguinarmy dovecot: imap-login: Login: user=<admin@penguinarmy.club>, method=PLAIN, rip=172.18.0.4, lip=172.18.0.10, mpid=8935, TLS, session=<4AKZ6r/mxNusEgAE>

Aug 21 09:00:35 penguinarmy postfix/submission/smtpd[8712]: connect from unknown[192.168.0.1]

Aug 21 09:00:35 penguinarmy postfix/submission/smtpd[8712]: Anonymous TLS connection established from unknown[192.168.0.1]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature ECDSA (P-384) server-digest SHA384

Aug 21 09:00:36 penguinarmy postfix/submission/smtpd[8712]: 121661ED28683: client=unknown[192.168.0.1], sasl_method=PLAIN, sasl_username=admin@penguinarmy.club

Aug 21 09:00:36 penguinarmy postfix/sender-cleanup/cleanup[8715]: 121661ED28683: message-id=<91ae9833-1cf2-8562-aebd-e9e5c7f8e288@penguinarmy.club>

Aug 21 09:00:36 penguinarmy postfix/sender-cleanup/cleanup[8715]: 121661ED28683: replace: header MIME-Version: 1.0 from unknown[192.168.0.1]; from=<admin@penguinarmy.club> to=<postmaster@penguinarmy.club> proto=ESMTP helo=<[10.13.13.2]>: Mime-Version: 1.0

Aug 21 09:00:36 penguinarmy opendkim[674]: 121661ED28683: DKIM-Signature field added (s=mail, d=penguinarmy.club)

Aug 21 09:00:36 penguinarmy postfix/qmgr[1948]: 121661ED28683: from=<admin@penguinarmy.club>, size=553, nrcpt=1 (queue active)

Aug 21 09:00:36 penguinarmy postfix/submission/smtpd[8712]: disconnect from unknown[192.168.0.1] ehlo=2 starttls=1 auth=1 mail=1 rcpt=1 data=1 quit=1 commands=8

Aug 21 09:00:37 penguinarmy dovecot: imap(admin@penguinarmy.club)<8935><4AKZ6r/mxNusEgAE>: Logged out in=832 out=1853 deleted=0 expunged=0 trashed=0 hdr_count=1 hdr_bytes=257 body_count=0 body_bytes=0

Aug 21 09:00:37 penguinarmy postfix/smtpd[8724]: connect from localhost[127.0.0.1]

Aug 21 09:00:37 penguinarmy postfix/smtpd[8724]: 81BF31ED28684: client=localhost[127.0.0.1]

Aug 21 09:00:37 penguinarmy postfix/cleanup[8725]: 81BF31ED28684: message-id=<91ae9833-1cf2-8562-aebd-e9e5c7f8e288@penguinarmy.club>

Aug 21 09:00:37 penguinarmy postfix/smtpd[8724]: disconnect from localhost[127.0.0.1] ehlo=1 mail=1 rcpt=1 data=1 quit=1 commands=5

Aug 21 09:00:37 penguinarmy postfix/qmgr[1948]: 81BF31ED28684: from=<admin@penguinarmy.club>, size=1652, nrcpt=1 (queue active)

Aug 21 09:00:37 penguinarmy postfix/trivial-rewrite[8665]: warning: do not list domain penguinarmy.club in BOTH mydestination and virtual_mailbox_domains

Aug 21 09:00:37 penguinarmy amavis[1969]: (01969-03) Passed CLEAN {RelayedInbound}, [192.168.0.1]:56263 <admin@penguinarmy.club> -> <admin@penguinarmy.club>, Queue-ID: 121661ED28683, Message-ID: <91ae9833-1cf2-8562-aebd-e9e5c7f8e288@penguinarmy.club>, mail_id: unNyl-HNwaoJ, Hits: 0.199, size: 1413, queued_as: 81BF31ED28684, 1048 ms

Aug 21 09:00:37 penguinarmy postfix/smtp[8716]: 121661ED28683: to=<admin@penguinarmy.club>, orig_to=<postmaster@penguinarmy.club>, relay=127.0.0.1[127.0.0.1]:10024, delay=1.6, delays=0.56/0/0/1, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 81BF31ED28684)

Aug 21 09:00:37 penguinarmy postfix/qmgr[1948]: 121661ED28683: removed

>>>>>>>>>>>>>>>>>>>>
Aug 21 09:00:37 penguinarmy postfix/local[8726]: 81BF31ED28684: to=<admin@penguinarmy.club>, relay=local, delay=0.15, delays=0.11/0/0/0.04, dsn=5.1.1, status=bounced (unknown user: "admin")
<<<<<<<<<<<<<<<<<<<<<<<

Aug 21 09:00:37 penguinarmy postfix/cleanup[8725]: A60CA1ED28683: message-id=<20220821130037.A60CA1ED28683@penguinarmy.club>

Aug 21 09:00:37 penguinarmy postfix/qmgr[1948]: A60CA1ED28683: from=<>, size=3502, nrcpt=1 (queue active)

Aug 21 09:00:37 penguinarmy postfix/bounce[8727]: 81BF31ED28684: sender non-delivery notification: A60CA1ED28683

Aug 21 09:00:37 penguinarmy postfix/trivial-rewrite[8665]: warning: do not list domain penguinarmy.club in BOTH mydestination and virtual_mailbox_domains

Aug 21 09:00:37 penguinarmy postfix/qmgr[1948]: 81BF31ED28684: removed

Aug 21 09:00:37 penguinarmy postfix/local[8726]: A60CA1ED28683: to=<admin@penguinarmy.club>, relay=local, delay=0.08, delays=0.04/0/0/0.04, dsn=5.1.1, status=bounced (unknown user: "admin")

Aug 21 09:00:37 penguinarmy postfix/qmgr[1948]: A60CA1ED28683: removed

Aug 21 09:00:43 penguinarmy dovecot: imap-login: Login: user=<admin@penguinarmy.club>, method=PLAIN, rip=172.18.0.4, lip=172.18.0.10, mpid=8972, TLS, session=<uqlF67/my9usEgAE>
```

**Comment by polarathene** [comment]
> Aug 21 09:00:37 penguinarmy postfix/trivial-rewrite[8665]: warning: do not list domain penguinarmy.club in BOTH mydestination and virtual_mailbox_domains

This is a misconfiguration. Share your config.

Did you also address the problem I caught in your previous discussion? Is `setup email list` still showing `admin@penguinarmy.club` having an alias to itself?

**Reply by JamesM92** [reply]
Yes, I've been removing all files and reinstalling periodically as i try to get this working

![image](https://user-images.githubusercontent.com/36548759/185928512-bc1c93a0-3f9e-43cb-9330-d22f23e3bf9c.png)


which config are you referring to? the docker compose file? or some config file inside the file directory?

**Reply by polarathene** [reply]
> which config are you referring to?

`docker-compose.yml`

**Reply by JamesM92** [reply]
```yml
version: "3"
#43

#mailserver tutorial
#1 -> https://docker-mailserver.github.io/docker-mailserver/edge/examples/tutorials/basic-installation/
#2 -> 

#lets encrypt guide
#https://docker-mailserver.github.io/docker-mailserver/edge/config/security/ssl/#lets-encrypt-recommended

#server setup checker
# https://mxtoolbox.com/emailhealth/penguinarmy.club/


networks:
  frontend:
    external: 
      name: FrontEnd
  backend:
    external:
      name: BackEnd

volumes:
  config:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/MailServer/_data/config
    
  data:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/MailServer/_data/data
 
  state:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/MailServer/_data/state
 
  log:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/MailServer/_data/logs
  
  certs: #the location of the certs is relative tot he certbot or nginx-proxy-manager that made them
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /srv/dev-disk-by-label-Archive/portainer/_certs/ssl
 
 
services:
  server:
    image: docker.io/mailserver/docker-mailserver:latest
    container_name: 03_MailServer
    restart: unless-stopped
    
    # If the FQDN for your mail-server is only two labels (eg: example.com),
    # you can assign this entirely to `hostname` and remove `domainname`.
    #hostname: mail
    #domainname: penguinarmy.club
    hostname: penguinarmy.club
    
    #env_file: stack.env
    environment:
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_CLAMAV=1
      - ENABLE_FAIL2BAN=1
      - ENABLE_POSTGREY=1
      - ENABLE_SASLAUTHD=0
      - ONE_DIR=1
      - SSL_TYPE=letsencrypt
    volumes:
      - data:/var/mail/
      - state:/var/mail-state/
      - log:/var/log/mail/
      - config:/tmp/docker-mailserver/
      - certs:/etc/letsencrypt/
      - /etc/localtime:/etc/localtime:ro
    #command:
    ports:
      - "25:25"    # SMTP  (explicit TLS => STARTTLS)
      #- "143:143"  # IMAP4 (explicit TLS => STARTTLS)
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
      #- "993:993"  # IMAP4 (implicit TLS)
    cap_add:
      - NET_ADMIN #needed for fail2ban to block ip's
    networks:
      - frontend
      - backend
    stop_grace_period: 1m
    healthcheck:
      test: "ss --listening --tcp | grep -P 'LISTEN.+:smtp' || exit 1"
      timeout: 3s
      retries: 0
```

**Reply by JamesM92** [reply]
Is there something more that is needed?

**Reply by polarathene** [reply]
> Is there something more that is needed?

No, that's good thanks.

I've just been rather busy and not sure when I would have time to investigate. It might help if you file a bug report (_remember to share your `docker-compose.yml` and logs_).

---

I do have one suggestion that should fix this. **Try `hostname: mail.penguinarmy.club`** (_and update your DNS records to support that_). 

The `hostname` should ideally use a sub-domain for your mail server, some of our configuration uses that value and doesn't have good compatibility with bare domains (eg: `example.com`).

Often there is a misunderstanding by users that setting the `hostname` like this must means email accounts become `user@mail.example.com`, but that is not true, it will remain `user@example.com`.

See the collapsed explanation below for more details if you're interested in understanding the issue better.

---

<details>
<summary>Explanation for why this is happening</summary>

I am rather certain that your problem is caused by the log line I pointed out:

> `Aug 21 09:00:37 penguinarmy postfix/trivial-rewrite[8665]: warning: do not list domain penguinarmy.club in BOTH mydestination and virtual_mailbox_domains`

- `mydestination` (_Postfix uses for local/internal mail from system accounts_)
- `virtual_mailbox_domains` (_Is for your configured mail accounts with real domains that deliver to Dovecot_)

---

The `hostname` is meant to tell Postfix that this is the address it uses as a mail server when it sends mail to other servers. It also affects configuring `mydestination` to use that value, which I believe is used internally for local system users (eg: `root`, `amavis`, `clamav`, etc). 

For email accounts you add to `postfix-accounts.cf` (eg with `setup email add`), Dovecot is configured to handle those domains (as _virtual mailbox domains_), it should not conflict with the Postfix `mydestination` (_which is for handling local mail internally unrelated to Dovecot_).

What I believe happened is Postfix thought you meant this mail belongs to the `admin` system user (doesn't exist), instead of handing it to `admin@penguinarmy.club` managed by Dovecot. This would be due to `mydestination` being configured for `@penguinarmy.club`, [which is not what we want](https://www.postfix.org/BASIC_CONFIGURATION_README.html#mydestination) (_we want to deliver to Dovecot virtual mailbox domain `@penguinarmy.club` instead_).

---

If we detect more than 1 "dot" in the `hostname`, we also configure `HOSTNAME` and `DOMAINNAME` ENV (_uses the `hostname` but ignores the first subdomain / left-most value_):

- `mail.example.com` => `example.com`
- `mail.example.co.uk` => `example.co.uk`
- `mail.testing.example.com` => `testing.example.com`

The value after arrows is what would be used for setting `DOMAINNAME`, and the removed left-side part is assigned to `HOSTNAME`. This does affect how services like Postfix get configured. That doesn't work well when using bare domains like `hostname: example.com`.

</details>

---

If you really can't use `mail.penguinarmy.club` let me know. We do have a workaround to support this, but I don't like to advise it, since it's rarely needed and often due to a misunderstanding by the user.
